content/blog/2022/2022-01-28-getting-started-with-amazon-kinesis.md

```diff
@@ -0,0 +1,858 @@
+Streaming data includes a wide variety of data such as log files generated by customers using mobile or web applications, e-commerce purchases, in-game player activity, information from social networks, financial trading floors, or geospatial services, and telemetry from connected devices or instrumentation in data centers.
+
+Streaming data is processed sequentially and incrementally either by one record at a time or in batches of records aggregated over sliding time windows. 
+
+## What is Amazon Kinesis
+Amazon Kinesis is a fully managed streaming data platform for processing streaming data. It provides four specialized services based on the type of stream data processing we want to perform to suit our use case:
+
+- **Kinesis Data Streams**: The Kinesis Data Streams service is used to capture streaming data produced by various data sources in real-time. Producer applications write to the Kinesis Data Stream and consumer applications connected to the stream read the data for different types of processing.
+
+- **Kinesis Data Firehose**: With Kinesis Data Firehose, we do not need to write applications or manage resources. We configure data producers to send data to Kinesis Data Firehose, and it automatically delivers the data to the specified destination. We can also configure Kinesis Data Firehose to transform the data before delivering it.
+
+- **Kinesis Data Analytics**:With Amazon Kinesis Data Analytics we can process and analyze streaming data. It provides an efficient and scalable environment to run Flink applications.
+
+- **Kinesis Video Streams**: Amazon Kinesis Video Streams is a fully managed AWS service that we can use to stream live media from video or audio capturing devices to the AWS Cloud, or build applications for real-time video processing or batch-oriented video analytics.
+
+Let us understand these services in the next sections. In each section, we will first introduce the key concepts of the service and then work through some examples.
+
+## Kinesis Data Streams - Ingesting and Storing Streaming Data 
+
+The Kinesis Data Streams service is used to collect and process streaming data as soon as it is produced (in real-time). The ingested data is emitted to another data store after some transformation or used to run any processing task for producing real-time metrics and analytics. The data is stored for 24 hours by default which can be configured to store the data beyond 24 hours up to 365 days.
+
+A common use of Kinesis Data Streams is the real-time aggregation of data followed by loading the aggregated data into a data warehouse or map-reduce cluster.
+
+### Kinesis Data Stream, Shards, and Records
+When using Kinesis Data Streams, we first set up a data stream and then build producer applications that write data to the data stream and consumer applications that read and process the data from the data stream in real-time:
+
+{{% image alt="Create Kinesis Delivery Stream" src="images/posts/aws-kinesis/kds-shards.png" %}}
```

Comment:
Have a look at https://excalidraw.com/ next time you create diagrams. The diagrams are a bit more attractive.